source/opt/dead_branch_elim_pass.cpp

```diff
@@ -73,260 +73,268 @@ bool DeadBranchElimPass::GetConstInteger(uint32_t selId, uint32_t* selVal) {
 }
 
 void DeadBranchElimPass::AddBranch(uint32_t labelId, ir::BasicBlock* bp) {
+  assert(get_def_use_mgr()->GetDef(labelId) != nullptr);
   std::unique_ptr<ir::Instruction> newBranch(new ir::Instruction(
       context(), SpvOpBranch, 0, 0,
       {{spv_operand_type_t::SPV_OPERAND_TYPE_ID, {labelId}}}));
   get_def_use_mgr()->AnalyzeInstDefUse(&*newBranch);
   bp->AddInstruction(std::move(newBranch));
 }
 
-void DeadBranchElimPass::AddSelectionMerge(uint32_t labelId,
-                                           ir::BasicBlock* bp) {
-  std::unique_ptr<ir::Instruction> newMerge(new ir::Instruction(
-      context(), SpvOpSelectionMerge, 0, 0,
-      {{spv_operand_type_t::SPV_OPERAND_TYPE_ID, {labelId}},
-       {spv_operand_type_t::SPV_OPERAND_TYPE_LITERAL_INTEGER, {0}}}));
-  get_def_use_mgr()->AnalyzeInstDefUse(&*newMerge);
-  bp->AddInstruction(std::move(newMerge));
+ir::BasicBlock* DeadBranchElimPass::GetParentBlock(uint32_t id) {
+  return context()->get_instr_block(get_def_use_mgr()->GetDef(id));
 }
 
-void DeadBranchElimPass::AddBranchConditional(uint32_t condId,
-                                              uint32_t trueLabId,
-                                              uint32_t falseLabId,
-                                              ir::BasicBlock* bp) {
-  std::unique_ptr<ir::Instruction> newBranchCond(new ir::Instruction(
-      context(), SpvOpBranchConditional, 0, 0,
-      {{spv_operand_type_t::SPV_OPERAND_TYPE_ID, {condId}},
-       {spv_operand_type_t::SPV_OPERAND_TYPE_ID, {trueLabId}},
-       {spv_operand_type_t::SPV_OPERAND_TYPE_ID, {falseLabId}}}));
-  get_def_use_mgr()->AnalyzeInstDefUse(&*newBranchCond);
-  bp->AddInstruction(std::move(newBranchCond));
-}
+bool DeadBranchElimPass::MarkLiveBlocks(
+    ir::Function* func, std::unordered_set<ir::BasicBlock*>* live_blocks) {
+  // |continues| maps the continue target to its corresponding header.
+  std::unordered_map<ir::BasicBlock*, ir::BasicBlock*> continues;
```

Comment:
Why is this a map?  Looks like a set would do.

Reply:
You're right. I think I originally made it a map assuming I'd need the mapped value, but never ended up using it.